kudoctl/src/config.rs

```diff
@@ -0,0 +1,116 @@
+use std::{
+    env,
+    fs::{create_dir_all, File},
+    path::{Path, PathBuf},
+};
+
+use log::LevelFilter;
+use serde::{Deserialize, Serialize};
+
+// Defaults configuration values
+
+fn default_log_level_str() -> String {
+    "info".to_string()
+}
+fn default_controller_url() -> String {
+    "http://localhost:8080".to_string()
+}
+fn default_config_file() -> PathBuf {
+    dirs::home_dir()
+        .unwrap_or(Path::new(".").to_path_buf())
+        .join(".kudo")
+        .join("config.yaml")
+}
+
+// Returns the right LevelFilter for the given log level string.
+fn get_verbosity_level_from_string(verbosity_level_str: &str) -> LevelFilter {
+    match verbosity_level_str {
+        "off" => LevelFilter::Off,
+        "error" => LevelFilter::Error,
+        "warn" => LevelFilter::Warn,
+        "info" => LevelFilter::Info,
+        "debug" => LevelFilter::Debug,
+        _ => LevelFilter::Info,
+    }
+}
+
+// Serializable configuration struct
+#[derive(Debug, Deserialize, Serialize)]
+pub struct ConfigFile {
+    #[serde(default = "default_controller_url")]
+    controller_url: String,
+    #[serde(default = "default_log_level_str")]
+    verbosity_level: String,
+}
+
+// This struct contains the configuration for the application.
+pub struct Config {
+    config_file: PathBuf,
+    pub controller_url: String,
+    pub verbosity_level: LevelFilter,
+}
+
+// Read the config file and return a Config object.
+// If the file does not exist, creates one with the default values.
+fn read_config_file(path: &PathBuf) -> Result<ConfigFile, Box<dyn std::error::Error>> {
+    if !path.exists() {
```

Comment:
The following construct is more idiomatic:

```rust
let config_file = if !path.exists() {
    ...
    Default::default()
} else {
     let file = File::open(path)?;
     serde_yaml::from_reader(file)?
};
```

Reply:
I tried to make it more idiomatic like you asked, but I'm not sure if the way the default config is written is good enough.